### DISCUSSION TITLE
migrating from v 12 to v 13

### DISCUSSION THREAD

**Discussion by codetojoy** [ebean-orm/ebean#3403] [General]
I am migrating a Play Framework app that is effectively moving from Ebean 12.16.1 to Ebean 13.17.3 (according to the doc for Play plugins).

There is a `findList()` query against a view in Postgres that now generates a NullPointerException in `DefaultServer.buildQueryRequest()` (line 941): "cannot invoke SpiQuery.setType() because spiQuery is null".

Apologies: I know this is absurdly vague, but I thought I would start here for help. I can't post the code or stack-trace (a brief one is at [1]). I can try to get a minimal example, but that could be difficult. Has anyone encountered this error, by chance?

[1] summary of stack-trace - 

```
SpiQuery.setType()
DefaultServer.buildQueryRequest()
DefaultServer.createQueryRequest()
DefaultServer.compileQuery()
SubQueryExpression.compileSubQuery()
SubQueryExpression.prepareExpression()
DefaultExpressionList.prepareExpression()
DefaultOrmQuery.prepareExpressions()
DefaultOrmQuery.prepare()
OrmQueryRequest.prepareQuery()
DefaultServer.findList()
```

**Comment by rob-bygrave** [comment]
> Has anyone encountered this error, by chance?

It does not ring any bells for me.

Note that it looks like the issue is in preparing/compiling the sub-query. In terms of reproducing the issue you can probably focus on the sub-query part.

https://github.com/ebean-orm/ebean/blob/13.17.3/ebean-core/src/main/java/io/ebeaninternal/server/core/DefaultServer.java#L941

**Reply by rob-bygrave** [reply]
Probably an issue with a sub-query created via https://github.com/ebean-orm/ebean/blob/13.17.3/ebean-core/src/main/java/io/ebeaninternal/server/expression/SubQueryExpression.java#L29

**Reply by codetojoy** [reply]
Thank you: your input helped me (a relative newbie)! The code was using an expression of `.ne("foobar", null)`. Using `.isNotNull("foobar")` works great to correct it.

In case it helps others: I believe that v 12 was silent (or that the query was not being executed), and that something in v 13 is checking it. Either way, the fixes appear to be straight-forward and make sense.
